dropshot/src/extractor/header.rs

```diff
@@ -9,19 +9,28 @@ use schemars::JsonSchema;
 use serde::de::DeserializeOwned;
 
 use crate::{
-    from_map::from_map, ApiEndpointBodyContentType,
+    from_map::from_map_insensitive, ApiEndpointBodyContentType,
     ApiEndpointParameterLocation, HttpError, RequestContext, RequestInfo,
     ServerContext,
 };
 
 use super::{metadata::get_metadata, ExtractorMetadata, SharedExtractor};
 
 /// `Header<HeaderType>` is an extractor used to deserialize an instance of
-/// `HeaderType` from an HTTP request's header values. `PathType` may be any
+/// `HeaderType` from an HTTP request's header values. `HeaderType` may be any
 /// structure that implements [serde::Deserialize] and [schemars::JsonSchema].
 /// While headers are accessible through [RequestInfo::headers], using this
 /// extractor in an entrypoint causes header inputs to be documented in
 /// OpenAPI output. See the crate documentation for more information.
+///
+/// Note that (unlike the [`Query`] and [`Path`] extractors) headers are case-
+/// insensitive. You may rename fields with mixed casing (e.g. by using
+/// #[serde(rename = "X-Header-Foo")]) and that casing will appear in the
+/// OpenAPI document output. Case-insensitive name conflicts may lead to
+/// unexpected behavior, and should be avoided. For example, only one of the
```

Comment:
Trying to make sense of "Case-insensitive name conflicts may lead to unexpected behavior, and should be avoided".

Who is this advice for? Why not just say "Since HTTP headers are case-insensitive, we automatically treat headers as lowercase. Upper-case characters are still allowed for OpenAPI documents, but they are converted to lowercase when deserializing and dealing with extraction conflicts".